libbeat/docs/loggingconfig.asciidoc

```diff
@@ -94,6 +94,21 @@ ifndef::serverless[]
 When true, writes all logging output to standard error output. This is
 equivalent to using the `-e` command line option.
 
+ifndef::serverless[]
+[float]
+==== `logging.with_fields`
+
+A map of key-value pair added to beat process logs for additional context.
+
+Usage:
+```
+logging.with_fields:
+  id: 123
+  data_stream:
+    dataset: beat
+    namespace: default
+```
```

Comment:
I think we can document this with type information like it is done below. Unfortunately I don't know what the type will look like, we might have to find some other config that allows arbitrary maps so we can copy the type information 😅. Unless someone more experienced with beats know this from the top of their head.